### PR TITLE
Add logo keyword to Documenter

### LINKED ISSUE #221: How to specify logo for use by Documenter plugin?
specifically, i believe a logo is an assets that should _not_  get added to the `HTML` in `makedocs` in `docs/make.jl`

- i'd like to include a `.png` logo in the docs
- to do this i think we need
  1. the `png` to exist  in `docs/src/assets/logo.png`
  2. the `.png` must  _not_ be mention in the `assets` in`makedocs(; format=Documenter.HTML(; assets=[...]), ...)` (as this seems to make Documenter unhappy, and causes `make.jl` to error with `ERROR: LoadError: Unrecognised asset class unknown for `assets/logo.png`. I think this is intended behaviour.)

Using `Documenter{GitLabCI}(assets=["foo.ext"])` adds `"foo.ext"` to `assets` in `HTML`/`makedocs`.

Quoting the [Documenter.HTML  docs](https://juliadocs.github.io/Documenter.jl/dev/lib/internals/writers/#Documenter.Writers.HTMLWriter.HTML):
> For the logo, Documenter checks for the existence of assets/logo.{svg,png,webp,gif,jpg,jpeg}, in this order. The first one it finds gets displayed at the top of the navigation sidebar. It will also check for assets/logo-dark.{svg,png,webp,gif,jpg,jpeg} and use that for dark themes.

### PR DESCRIPTION
**Pull request by christopher-dG**
Closes #221

I think this does the right thing, at least.

### CODE REVIEW

**Comment by nickrobinson251** [comment]
Nice! Thanks! This PR looks perfect for the assumption there's a single logo (andis sufficient for my use-case), however Documenter actually supports have two logos (light theme and dark theme) - how should that case be handled?

> Documenter checks for the existence of assets/logo.{svg,png,webp,gif,jpg,jpeg}, in this order. The first one it finds gets displayed at the top of the navigation sidebar. **It will also check for assets/logo-dark.{svg,png,webp,gif,jpg,jpeg} and use that for dark themes.**

**Comment by christopher-dG** [comment]
ah shucks

another kw `logo_dark` I guess?

**Review by nickrobinson251 — APPROVED** [review]
Changes LGTM! Although see question about `logo-dark.png` case :)

**Comment by nickrobinson251** [comment]
> another kw `logo_dark` I guess?

🤷‍♂️ 

sure 😆

File: src/plugins/documenter.jl [hunk 1]
```diff
@@ -59,12 +62,15 @@ end
...
     logo::Union{AbstractString, Nothing}=nothing,
+    logo_dark::Union{AbstractString, Nothing}=nothing,
     makedocs_kwargs::Dict{Symbol}=Dict{Symbol, Any}(),
     canonical_url::Union{Function, Nothing}=make_canonical(T),
     make_jl::AbstractString=default_file("docs", "make.jl"),
     index_md::AbstractString=default_file("docs", "src", "index.md"),
 ) where T <: DeployStyle
-    return Documenter{T}(assets, logo, makedocs_kwargs, canonical_url, make_jl, index_md)
+    return Documenter{T}(
+        assets, logo, logo_dark,makedocs_kwargs, canonical_url, make_jl, index_md,
```
Comment (nickrobinson251) [inline src/plugins/documenter.jl:72]:
```suggestion
        assets, logo, logo_dark, makedocs_kwargs, canonical_url, make_jl, index_md,
```

**Review by nickrobinson251 — APPROVED** [review]
thanks again!

**Comment by christopher-dG** [comment]
I feel like 826aa536adbea87f7ddbc382a3ba5cffcf9733e6 is maybe a bit better... thoughts?

**Pull request merged**